source/js/GliderRecipe.js

```diff
@@ -0,0 +1,201 @@
+class GliderRecipe extends HTMLElement {
+  constructor(){
+    super();
+    let shadow = this.attachShadow({mode: 'open'});
+  }
+
+  set data(data) {
+
+    let cleanData = {};
+    /** 
+     * Making a more organized object for getting 
+     * the info from messy recipe json. Might not end up using all of it,
+     * I just yoinked this from my lab code
+     * 
+     * cleanData should have
+     * cleanData.thumbnail - link to the photo of recipe
+     * cleanData.title - title of the recipe
+     * cleanData.url - url for the recipe
+     * cleanData.organization - organization of the recipe
+     * cleanData.time - cook time of the recipe
+     * cleanData.ingredients - string of ingredients
+     * 
+     * cleanData.rating is optional and should be
+     * rating.score - review score out of 5
+     * rating.count - number of reviews
+     */
+
+    // process raw data into cleanData
+    cleanData.thumbnail = searchForKey(data,'thumbnailUrl');
+    cleanData.title = getRecipeTitle(data);
+    cleanData.url = getUrl(data);
+    cleanData.organization = getOrganization(data);
+    let tempTime = searchForKey(data,'cookTime');
+    if (!tempTime) tempTime = searchForKey(data,'totalTime');
+    cleanData.time = convertTime(tempTime);
+    cleanData.ingredients = createIngredientList(searchForKey(data,'recipeIngredient'));
+    let tempRating = searchForKey(data,'aggregateRating');
+    if (tempRating) {
+      cleanData.rating = {
+        count: tempRating.ratingCount,
+        score: tempRating.ratingValue
+      }
+    }
+
+    // alright cool we got everything, time to actually build it
+
+    // this is gonna be the root element we'll slap everything else onto
+    const entry = document.createElement('li');
+    entry.classList.add('glide__slide');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8447291)
```suggestion
    entry.classList.add("glide__slide");
```